.github/workflows/end2end-tests.yml

```diff
@@ -25,8 +25,30 @@ jobs:
       - name: Build
         run: go build -v
 
-      - name: Run
-        run: ./GO4SQL -file test_file > output.txt
-
-      - name: Check Result
-        run: diff output.txt ./.github/expected_results/end2end.txt
+      - name: Run Tests
+        run: |
+          e2e_failed=false
+          
+          for test_file in e2e/test_files/*_test; do
+            output_file="./e2e/$(basename "${test_file/_test/_output}")"
+          
+            ./GO4SQL -file "$test_file" > "$output_file"
+          
+            expected_output="e2e/expected_outputs/$(basename "${test_file/_test/_expected_output}")"
+          
+            diff "$output_file" "$expected_output"
+          
+            if [ $? -ne 0 ]; then
+              echo "E2E test for: {$test_file} failed"
+              e2e_failed=true
+            fi
+          
+            rm "./$output_file"
+          done
+          
+          if [ "$e2e_failed" = true ]; then
+            echo "E2E tests failed."
+            exit 1
+          else
+            echo "All E2E tests passed."
+          fi
```

Comment:
Can we do not keep scripts like that in our workflows? It would be much better just to run bash script instead of whole bash block of code.